Make Archives Block display correct empty message

caxco93 opened the pull request:
## Description
When creating an Archives Block while having no posts published, the
Archives Block is empty and overlaps with the subsequent Blocks.

This addresses that by setting the correct Block content when there
are no Archives to show.

## How has this been tested?
Built in tests and manually

## Screenshots
*Before:*
![emptyarchives](https://user-images.githubusercontent.com/3190666/44760513-b1712000-ab04-11e8-866f-fd75cb2989ed.gif)

*After:*
![emptyarchives2](https://user-images.githubusercontent.com/3190666/44760520-b635d400-ab04-11e8-98ba-09854b1051ed.gif)

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [ ] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->

Fixes #9418

jorgefilipecosta commented on packages/block-library/src/archives/index.php:
```diff
@@ -79,12 +79,16 @@ function render_block_core_archives( $attributes ) {
+			$archives
+		) : sprintf(
+			'<div class="%1$s">%2$s</div>',
+			$classnames,
```
We are outputting an empty class on the frontend `<div class="">No archives to show.</div>`. Should we use `esc_attr( $class )` here instead of `$classnames` ?

caxco93 replied:
Thanks for noticing!

caxco93 committed:
Make Archives Block display correct empty message

When creating an Archives Block while having no posts published, the
Archives Block is empty and overlaps with the subsequent Blocks.

This addresses that by setting the correct Block content when there
are no Archives to show.

pento commented:
@caxco93: Thanks for the PR!

It seems like this could be simplified further: rather than duplicating the `sprintf()` call, it could be something like:

```php
		$block_content = sprintf(
			'<ul class="%1$s">%2$s</ul>',
			esc_attr( $class ),
			$archives ? $archives : __( 'No archives to show.', 'gutenberg' )
		);
```

caxco93 committed:
Changed ternary for `if` to increase readability

pento commented:
Oh, I totally missed the `<ul>` / `<div>` difference. 🙂

This looks good to me, thanks @caxco93!

karmatosed commented:
Just a little note, we probably should have had a design check on this but we can retro do. In future as this is a state lets get insight here.
